src/io/oss_io.ml

```diff
@@ -71,15 +73,9 @@ object (self)
           Unix.close x ;
           fd <- None
 
-  method output_start =
-    if clock_safe then
-      (get_clock ())#register_blocking_source ;
```

Comment:
Why don't you have to register/unregister anymore?

Reply:
That was the way for a source to declare that it had self-synchronization capabilities. It was done globally when waking up and did not account for wether or not that source was actually being pulled. This mechanism has been brought down to the frame producing level and accounts for the source that is actually filling the frame.